cadc-util/src/intTest/java/ca/nrc/cadc/db/version/KeyValueDAOTest.java

```diff
@@ -0,0 +1,210 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+public class KeyValueDAOTest {
+    private static final Logger log = Logger.getLogger(KeyValueDAOTest.class);
+
+    private static final String SCHEMA = "dbversion";
+    private static final String DATABASE = "cadctest";
+    private static final String MODELNAME = "KeyValueDAOTest";
+    private static final String MODELVERSION = "0.1";
+    private static final String TABLE_NAME = SCHEMA + ".KeyValue";
+    private static DataSource dataSource;
+
+    private static final String CREATE = "create table dbversion.KeyValue " +
+            "(name varchar(32) not null primary key,value varchar(32) not null,lastModified timestamp not null)";
+
+    public KeyValueDAOTest() {
+    }
+
+    @BeforeClass
+    public static void setup() {
+        Log4jInit.setLevel("ca.nrc.cadc.db.version", Level.DEBUG);
+
+        try {
+            DBConfig dbrc = new DBConfig();
+            ConnectionConfig cc = dbrc.getConnectionConfig("KEYVALUE_DAO_TEST", DATABASE);
+            DBUtil.createJNDIDataSource("jdbc/KeyValueDAOTest", cc);
+            dataSource = DBUtil.findJNDIDataSource("jdbc/KeyValueDAOTest");
+
+            DatabaseTransactionManager txn = new DatabaseTransactionManager(dataSource);
+            JdbcTemplate jdbc = new JdbcTemplate(dataSource);
+
+            try {
+                jdbc.execute(String.format("DROP TABLE %s", TABLE_NAME));
+            } catch (Exception e) {
+                log.warn("drop failed: " + e);
+            }
+
+            try {
+                jdbc.execute(CREATE);
+            } catch (Exception e) {
+                log.error("create failed: " + e);
+                Assert.fail("create failed: " + e);
+            }
+        } catch (Exception e) {
+            log.error("unexpected exception", e);
+            Assert.fail("unexpected exception: " + e);
+        }
+    }
+
+    @Before
+    public void cleanup() throws Exception {
+        String sql = String.format("DELETE FROM %s", TABLE_NAME);
+        log.info("cleanup: "  + sql);
+        try {
+            dataSource.getConnection().createStatement().execute(sql);
+        } catch (SQLException oops) {
+            log.warn("delete failed: " + oops);
+        }
+    }
+
+    @Test
+    public void testCreateGetDelete() {
+        try {
+            KeyValueDAO dao = new KeyValueDAO(dataSource, DATABASE, SCHEMA);
+
+            KeyValue expected = new KeyValue("test-key");
+            expected.value = "test-value";
+            dao.put(expected);
+
+            KeyValue actual = dao.get(expected.getName());
+            Assert.assertNotNull(actual);
+            Assert.assertEquals(expected.getName(), actual.getName());
+            Assert.assertEquals(expected.value, actual.value);
+            Assert.assertEquals(expected.lastModified, actual.lastModified);
+
+            try {
```

Comment:
Why this extra try/catch?

Reply:
I guess to give a better error message about where the failure occurred, but it could've failed multiple times before then, so likely not terribly helpful. Removed.